c/heap-manipulation/tree_false-unreach-call_true-valid-memsafety.c

```diff
@@ -52,6 +52,8 @@ struct node *create_tree()
         node->value = value;
         nodelast = node;
     }
+    if (node != NULL)
+      node->parent = NULL;
```

Comment:
Which appears to imply the the previous/current code actually isn't memory safe!? (With the assumption that dynamically allocated memory has non-deterministic contents, nodelast->parent is non-deterministic, and thus not necessarily NULL. Hence and invalid pointer would be dereferenced in line 75.)